client/rest.go

```diff
@@ -0,0 +1,259 @@
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ethcmn "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/gorilla/mux"
+	"math/big"
+	"net/http"
+)
+
+func RegisterRoutes(ctx context.CLIContext, r *mux.Router) {
+	r.HandleFunc("/balance/{address}", balanceHandler(ctx)).Methods("GET")
+	r.HandleFunc("/info/{address}", infoHandler(ctx)).Methods("GET")
+	r.HandleFunc("/block/{height}", blockHandler(ctx)).Methods("GET")
+	r.HandleFunc("/blocks/{height}", blocksHandler(ctx)).Methods("GET")
+	r.HandleFunc("/tx/{hash}", txHandler(ctx)).Methods("GET")
+	r.HandleFunc("/output/{position}", outputHandler(ctx)).Methods("GET")
+	r.HandleFunc("/submit", submitHandler(ctx)).Methods("POST")
+}
+
+func balanceHandler(ctx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		addr := vars["address"]
+		if !ethcmn.IsHexAddress(addr) {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("address must be an ethereum 20-byte hex string"))
+			return
+		}
+
+		total, err := Balance(ctx, ethcmn.HexToAddress(addr))
+		if err != nil {
+			sdkerr, ok := err.(sdk.Error)
+			if !ok || sdkerr.Code() != store.CodeDNE {
+				w.WriteHeader(http.StatusBadRequest)
+			} else {
+				w.WriteHeader(http.StatusOK)
+			}
+
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(total))
+	}
+}
+
+func infoHandler(ctx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		addr := vars["address"]
+		if !ethcmn.IsHexAddress(addr) {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("address must be an ethereum 20-byte hex string"))
+			return
+		}
+
+		txo, err := Info(ctx, ethcmn.HexToAddress(addr))
+		if err != nil {
+			sdkerr, ok := err.(sdk.Error)
+			if !ok || sdkerr.Code() != store.CodeDNE {
+				w.WriteHeader(http.StatusBadRequest)
+			} else {
+				w.WriteHeader(http.StatusOK)
+			}
+
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		data, err := json.Marshal(txo)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+		w.Write(data)
+	}
+}
+
+func blockHandler(ctx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		num, ok := new(big.Int).SetString(mux.Vars(r)["height"], 10)
+		if !ok || num.Sign() <= 0 {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("block height must be in decimal format starting from 1"))
+			return
+		}
+
+		block, err := Block(ctx, num)
+		if err != nil {
+			sdkerr, ok := err.(sdk.Error)
+			if !ok || sdkerr.Code() != store.CodeDNE {
+				w.WriteHeader(http.StatusBadRequest)
+			} else {
+				w.WriteHeader(http.StatusOK)
+			}
+
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		resp := struct {
+			store.Block
+			Txs [][]byte
+		}{
```

Comment:
```suggestion
		}{}
```
can't you just do `{}`

Reply:
yes, you can. Just tested in go playground. Had the wrong assumption that `append` did not work with nil slices.